src/Routing/FrontendLoader.php

```diff
@@ -85,16 +73,8 @@ public function supports($resource, $type = null)
      */
     private function addFrontendRoute(RouteCollection $routes, array $defaults)
     {
-        $pattern = '/{alias}';
-        $require = ['alias' => '.*'];
-
-        // URL suffix
-        if ($this->format) {
-            $pattern .= '.{_format}';
-
-            $defaults['_format'] = $this->format;
-            $require['_format']  = $this->format;
-        }
+        $pattern = '/{alias}%contao.url_suffix%';
+        $require = ['alias' => '.+'];
```

Comment:
That's correct, otherwise the URL would be `de/.html` if you provide no alias with a german locale.